<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1045</width>
    <height>951</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1045</width>
    <height>951</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1045</width>
    <height>951</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>3D Viewer</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="open_button">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>5</y>
      <width>211</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Open</string>
    </property>
   </widget>
   <widget class="QFrame" name="changes">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>211</width>
      <height>361</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="translation_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Translation</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotation_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>120</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotation</string>
     </property>
    </widget>
    <widget class="QLabel" name="scalling_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>240</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Scalling</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="scale_x_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>268</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="scale_y_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>298</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="scale_z_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>328</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="scale_pushButton">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>265</y>
       <width>61</width>
       <height>88</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
    <widget class="QLabel" name="scale_z_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>331</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLabel" name="scale_y_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>300</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="scale_x_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>270</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_x_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>148</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_y_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>178</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotate_y_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_z_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>208</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotate_z_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>211</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotate_x_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="translate_x_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>28</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="translate_y_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>58</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="translate_y_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="translate_z_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>88</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="translate_z_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>91</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLabel" name="translate_x_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rotate_pushButton">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>145</y>
       <width>61</width>
       <height>88</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
    <widget class="QPushButton" name="translate_pushButton">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>24</y>
       <width>61</width>
       <height>88</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="information">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>10</y>
      <width>571</width>
      <height>131</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="filename_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Filename:</string>
     </property>
    </widget>
    <widget class="QLabel" name="vertices_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>70</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertices:</string>
     </property>
    </widget>
    <widget class="QLabel" name="edges_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>100</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Edges:</string>
     </property>
    </widget>
    <widget class="QLabel" name="information_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Information</string>
     </property>
    </widget>
    <widget class="QLabel" name="filename_label_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>40</y>
       <width>551</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="vertices_label_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>70</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="edges_label_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>100</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </widget>
   <widget class="MyOpenGLWidget" name="opengl" native="true">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>150</y>
      <width>791</width>
      <height>791</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>791</width>
      <height>791</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>791</width>
      <height>791</height>
     </size>
    </property>
   </widget>
   <widget class="QFrame" name="projection">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>211</width>
      <height>61</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="projection_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Projection type</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="central_radio">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Central</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="parallel_radio">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Parallel</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="edge">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>510</y>
      <width>211</width>
      <height>121</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="edge_settings_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Edge settings</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="solid_radio">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Solid</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="dashed_radio">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Dashed</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="thickness_lineEdit">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>60</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="thickness_label">
     <property name="geometry">
      <rect>
       <x>24</x>
       <y>63</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Thickness</string>
     </property>
    </widget>
    <widget class="QPushButton" name="edge_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>84</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="vertex">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>640</y>
      <width>211</width>
      <height>141</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="vertex_settings_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertex settings</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="default_radio">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Default</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="circle_radio">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>53</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Circle</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="square_radio">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>53</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Square</string>
     </property>
    </widget>
    <widget class="QLabel" name="size_label">
     <property name="geometry">
      <rect>
       <x>24</x>
       <y>83</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Size</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="size_lineEdit">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>80</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QPushButton" name="vertex_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>104</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="color">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>10</y>
      <width>211</width>
      <height>131</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="color_settings_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Color settings</string>
     </property>
    </widget>
    <widget class="QPushButton" name="background_color_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Background color</string>
     </property>
    </widget>
    <widget class="QPushButton" name="edge_color_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Edge color</string>
     </property>
    </widget>
    <widget class="QPushButton" name="vertex_color_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertex color</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="save">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>790</y>
      <width>211</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="save_label">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Saving image</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="bmp_radio">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>.bmp</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="jpeg_radio">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>.jpeg</string>
     </property>
    </widget>
    <widget class="QPushButton" name="save_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>55</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Save image</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="gif">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>889</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="gif_pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Make gif</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">myopenglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
